File: ws/chiriboga/u1/ws08CDJavaClasses/ChickenFarmSimulator/src/ec/edu/espe/farmsimulator/model/ChickenFarmer.java
package ec.edu.espe.farmsimulator.model;


import java.util.List;

/**
 *
 * @autor Kerlly Chiriboga, ODS
 */
public class ChickenFarmer {
    private int id;
    private String name;
    private List<ChickenCoop> coops;

    public ChickenFarmer(int id, String name, List<ChickenCoop> coops) {
        this.id = id;
        this.name = name;
        this.coops = coops;
    }

    @Override
    public String toString() {
        StringBuilder coopsInfo = new StringBuilder();
        for (ChickenCoop coop : coops) {
            coopsInfo.append(coop.toString()).append("\n");
        }
        return "ChickenFarmer {" + id + "," + name + "} \n" + coopsInfo.toString();
    }
    
    public String toCSV() {
        StringBuilder sb = new StringBuilder();
        sb.append(id).append(",").append(name).append("\n");
        for (ChickenCoop coop : coops) {
            sb.append(coop.toCSV()).append("\n");
        }
        return sb.toString();
    }

    public void addCoop(ChickenCoop coop) {
        coops.add(coop);
    }

    public void removeCoop(ChickenCoop coop) {
        coops.remove(coop);
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<ChickenCoop> getCoops() {
        return coops;
    }

    public void setCoops(List<ChickenCoop> coops) {
        this.coops = coops;
    }
}

File: ws/chiriboga/u1/ws08CDJavaClasses/ChickenFarmSimulatorPython/farm_simulator/model/ChickenFarmer.py
# File: ec/edu/espe/farmsimulator/model/ChickenFarmer.py

from farm_simulator.model.ChickenCoop import ChickenCoop

class ChickenFarmer:
    def __init__(self, id, name, coops):
        self.id = id
        self.name = name
        self.coops = coops

    def __str__(self):
        coops_info = "\n".join([str(coop) for coop in self.coops])
        return f"ChickenFarmer {{ {self.id},{self.name} }} \n{coops_info}"

    def to_csv(self):
        csv_data = f"{self.id},{self.name}\n"
        csv_data += "\n".join([coop.to_csv() for coop in self.coops])
        return csv_data

    def add_coop(self, coop):
        self.coops.append(coop)

    def remove_coop(self, coop):
        self.coops.remove(coop)

    # Getters and setters omitted for brevity
